handlers/handlers_test.go

```diff
@@ -186,11 +250,76 @@ func defaultParams() url.Values {
 	values.Set("offset", "0")
 	values.Set("fromDate", "")
 	values.Set("toDate", "")
-	values.Set("sort", "release_date_desc")
+	values.Set("sort", queryparams.RelDateDesc.BackendString())
 	values.Set("keywords", "")
 	values.Set("query", "")
-	values.Set("release-type", "type-upcoming")
+	values.Set("release-type", queryparams.Upcoming.String())
 	values.Set("highlight", "true")
 
 	return values
 }
+
+func defaultICSParams() url.Values {
+	values := url.Values{}
+	values.Set("limit", "1000")
+	values.Set("toDate", time.Now().AddDate(0, 3, 0).Format(queryparams.DateFormat))
+	values.Set("sort", queryparams.RelDateAsc.BackendString())
+	values.Set("release-type", queryparams.Upcoming.String())
+
+	return values
+}
+
+func TestICalDate(t *testing.T) {
+	Convey("given a set of badly formatted date strings", t, func() {
+		bad := []string{"1st Jan 2020", "21-03-2021", "2021-03-04T12:10:00"}
+		Convey("verify the function simply returns an empty string", func() {
+			for _, d := range bad {
+				icd := iCalDate(context.Background(), d)
+				So(icd, ShouldEqual, "")
+			}
+		})
+	})
+
+	Convey("but given a set of well formatted date strings", t, func() {
+		good := []struct{ date, icalDate string }{
+			{date: "2021-03-04T12:10:00Z", icalDate: "20210304T121000Z"},
+			{date: "2021-03-04T12:10:00.000Z", icalDate: "20210304T121000Z"},
+			//{date: "2021-03-04T12:10:00Z05", icalDate: "20210304T071000Z"},
```

Comment:
Should it be able to handle timestamps with a timezone offset after all?

Reply:
I agree with Giles. It should work with this line active but the right value for `date` here would be something like `2021-03-04T12:10:00+05:00`

Reply:
I believe the strict answer to your question is no - it looks like all times stored in the releases are in UTC. This makes sense as it's a big mistake to store times in any other format (as opposed to displaying the time in a local zone).
Which is good news because time.Parse() will often give an error on some valid formats when it has a zone (yes indeed!) So this form the golang documentation:I meant to fix up the tescase above and this is it:
{date: "2021-03-04T12:10:00+05:00", icalDate: "20210304T071000Z"},
"2021-03-04T12:10:00Z05:00" is not regarded valid in time.Parse(time.RFC3339)